python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -556,8 +535,76 @@ def _plot_data(name, idx, marker_style=None):
         for type, info in _SOLUTION_TYPE_MAP.items():
             _plot_data(info.name, solution_type == type, marker_style=info.style)
 
-        self._add_figure(name="top_down", figure=topo_figure, title="Displacement: Top-Down (Topocentric)")
-        self._add_figure(name="displacement", figure=time_figure, title="Displacement: vs. Time")
+        name = source.replace(' ', '_').lower()
+        self._add_figure(name=f"{name}_top_down", figure=topo_figure, title=f"{source} Displacement: Top-Down (Topocentric)")
```

Comment:
I suggest moving "Displacement" to the name passed in by the pose displacement plot, and for rel pos setting the name to like "relative position" or something to avoid confusion. See comment below.